src/dune/build_system.ml

```diff
@@ -296,6 +303,10 @@ end = struct
   module Entry = struct
     type t =
       { rule_digest : Digest.t
+        (* We use Glob.t instead of Predicate.t for converting from
+          'dune_action' Dependency.t to Dep.t so we can safely marshall the
+           `dynamic_deps_stages`. *)
+      ; dynamic_deps_stages : (Dep.Set.t * Digest.t) list
```

Comment:
Could we use the type from `Protocol` here? I guess not because the paths are cwd-relative rather than repo-root-relative. I proposed in another PR to make another type for serializable `Dep.Set.t`, but I don't know. How much churn do you think that is?

Reply:
Introduced separate type for serializing dynamic dependencies required for dynamic run. (See `Action_exec.Dynamic_dep.t`)